fuse_optimizers/include/fuse_optimizers/fixed_size_smoother.h

```diff
@@ -0,0 +1,197 @@
+/*
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2019, Locus Robotics
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the copyright holder nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+#ifndef FUSE_OPTIMIZERS_FIXED_SIZE_SMOOTHER_H
+#define FUSE_OPTIMIZERS_FIXED_SIZE_SMOOTHER_H
+
+#include <fuse_core/graph.h>
+#include <fuse_core/transaction.h>
+#include <fuse_core/uuid.h>
+#include <fuse_optimizers/fixed_size_smoother_params.h>
+#include <fuse_optimizers/variable_stamp_index.h>
+#include <fuse_optimizers/windowed_optimizer.h>
+#include <ros/node_handle.h>
+#include <ros/time.h>
+
+#include <mutex>
+#include <string>
+#include <vector>
+
+namespace fuse_optimizers
+{
+/**
+ * @brief A fixed-Size smoother implementation that marginalizes out variables that are older than a defined Size time
```

Comment:
```suggestion
 * @brief A fixed-size smoother implementation that marginalizes out variables that are older than a defined size
```
Or maybe?
```suggestion
 * @brief A fixed-size smoother implementation that marginalizes out variables that are older than a defined buffer length
```

Also, it looks like find&replace got away from you. Review all of the comments and make sure `Size` is properly capitalized and makes sense in the sentence.